modules/http2/h2_mplx.c

```diff
@@ -896,8 +940,8 @@ static void s_c2_done(h2_mplx *m, conn_rec *c2, h2_conn_ctx_t *conn_ctx)
     ap_log_cerror(APLOG_MARK, APLOG_TRACE1, 0, c2,
                   "h2_mplx(%s-%d): c2 done", conn_ctx->id, conn_ctx->stream_id);
 
-    ap_assert(conn_ctx->done == 0);
-    conn_ctx->done = 1;
+    AP_DEBUG_ASSERT(apr_atomic_read32(&conn_ctx->done) == 0);
+    apr_atomic_set32(&conn_ctx->done, 1);
```

Comment:
TOCTOU, should this be:
    if (apr_atomic_cas32(&conn_ctx->done, 1, 0) != 0) {
        APR_DEBUG_ASSERT(0);
    }
?
And maybe `return;` if already set too? It looks like `s_c2_done()` is not supposed to be called multiple times and if it happens httpd's teeth will fall out (or can we simply ignore it)?

Reply:
Oh this is the only place where `->done` (so not a TOCTOU issue, no CAS needed)..
Probaly `return;` on already set is meaningful still, or it should be an `ap_assert()`?

Reply:
A DEBUG_ASSERT() seems best here. It is really only a dev check if things got screwed up by a change. This method needs to be called once and only once for a c2 connection.